### DISCUSSION TITLE
Code Formatter & Assist

### DISCUSSION THREAD

**Discussion by st11x** [AnWeber/vscode-httpyac#152] [Q&A]
Hi,

I can't select a block of JSON in the http file and format it as a JSON selection, the option is not available. Nor is formatting the entire document working. Is there a configuration missing that I am missing?

Also, in a script block {{ }}, JS code assist is not available as well. 

thanks
   Matt

**Comment by AnWeber** [comment]
No. It is not currently implemented. I had once looked at the logic of [vetur](https://github.com/vuejs/vetur/blob/40b1258ea3b02f61147610c816f57abea70f44f1/server/src/modes/script/javascript.ts#L698) which implemented this feature. But this has a certain complexity and in principle it would make sense to implement an LSP server (see #101) for performance reasons. My trick is to format JSON outside, or to save this directly into a separate file and [reference it](https://httpyac.github.io/guide/request.html#imported-request-body).
If you want to try your hand at the feature, feel free. I can possibly support you with implementing a PR or a custom extension. But I don't have the time to implement it myself right now. Sorry

**Reply by st11x** [reply]
Thanks, I appreciate your response on this. Yes, I moved some of the larger JSON outside of the file for this purpose; it is a good workaround.  I don't know if I am capable of implementing such a feature but I will take a look and let you know if I am up to it. 

Thanks!
